homeassistant/components/synology_dsm/camera.py

```diff
@@ -0,0 +1,97 @@
+"""Support for Synology DSM cameras."""
+from typing import Dict
+
+from synology_dsm.api.surveillance_station import SynoSurveillanceStation
+
+from homeassistant.components.camera import SUPPORT_STREAM, Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import SynologyDSMEntity
+from .const import (
+    DOMAIN,
+    ENTITY_CLASS,
+    ENTITY_ENABLE,
+    ENTITY_ICON,
+    ENTITY_NAME,
+    ENTITY_UNIT,
+    SYNO_API,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the Synology NAS binary sensor."""
+
+    api = hass.data[DOMAIN][entry.unique_id][SYNO_API]
+
+    if SynoSurveillanceStation.CAMERA_API_KEY not in api.dsm.apis:
+        return True
```

Comment:
Nothing is checking this return value. Just `return`.

Reply:
I created a PR for this (#39929).